packages/playwright/src/reporters/merge.ts

```diff
@@ -406,7 +427,20 @@ class IdsPatcher {
   }
 
   private _mapTestId(testId: string): string {
-    return this._stringPool.internString(testId + this._salt);
+    const t1 = this._stringPool.internString(testId);
+    if (this._testIdsMap.has(t1))
+      // already mapped
+      return this._testIdsMap.get(t1)!;
+    if (this._testIdsUsedInOtherBlobs.has(t1)) {
```

Comment:
It seems like you can have a single `globalTestIdSet` that you populate every time you assign a new unique id across blobs? Should be a bit easier than two sets.